packages/firestore/src/local/local_store.ts

```diff
@@ -704,49 +696,56 @@ export class LocalStore {
    * Notify local store of the changed views to locally pin documents.
    */
   notifyLocalViewChanges(viewChanges: LocalViewChanges[]): Promise<void> {
-    for (const viewChange of viewChanges) {
-      const targetId = viewChange.targetId;
-
-      this.localViewReferences.addReferences(viewChange.addedKeys, targetId);
-      this.localViewReferences.removeReferences(
-        viewChange.removedKeys,
-        targetId
-      );
-
-      if (!viewChange.fromCache) {
```

Comment:
Note that this part of the loop is just moved to below the transaction. The goal is that if the transaction fails, this code doesn't get executed.